spec/event_builder_spec.rb

```diff
@@ -34,216 +34,336 @@
+
+    experiment = config.get_experiment_from_key('test_experiment')
+    impression_event = @event_builder.create_impression_event(experiment, '111128', 'test_user', {invalid_attribute: 'sorry_not_sorry'})
+    expect(impression_event.params).to eq(@expected_impression_params)
+    expect(impression_event.url).to eq(@expected_event_builder_url)
     expect(impression_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called' do
     conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', nil, nil, {'111127' => '111128'})
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with attributes' do
-    @expected_conversion_params['userFeatures'] = [{
-      'id' => '111094',
-      'name' => 'browser_type',
-      'type' => 'custom',
-      'value' => 'firefox',
-      'shouldIndex' => true,
+    @expected_conversion_params[:visitors][0][:attributes] = [{
+      entity_id: '111094',
+      key: 'browser_type',
+      type: 'custom',
+      value: 'firefox',
     }]
 
     conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', {'browser_type' => 'firefox'}, nil, {'111127' => '111128'})
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with revenue event tag' do
-    @expected_conversion_params['eventMetrics'] = [{
-      'name' => 'revenue',
-      'value' => 4200,
-    }]
-    @expected_conversion_params['eventFeatures'] = [
-      {
-        'name' => 'revenue',
-        'type' => 'custom',
-        'value' => 4200,
-        'shouldIndex' => false
-      },
-    ]
+    event_tags = { 'revenue' => 4200 }
 
-    event_tags = {'revenue' => 4200}
+    @expected_conversion_params[:visitors][0][:attributes] = []
+    @expected_conversion_params[:visitors][0][:snapshots][0][:events][0].merge!({
+      revenue: 4200,
+      tags: event_tags
+    })
 
     conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', nil, event_tags, {'111127' => '111128'})
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with revenue event tag' do
```

Comment:
Is it a duplicate of the one on Line 175?

Reply:
no, both are different